apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -890,14 +891,22 @@ func (ctx *prepareOrGetContext) ProcessOneContainer(env environs.Environ, callCo
 	// into things we'd like to tell the Host machine to create, and then *it*
 	// reports back what actually exists when its done.
 	if err := bridgePolicy.PopulateContainerLinkLayerDevices(host, container); err != nil {
-		return err
+		return errors.Trace(err)
 	}
 
 	containerDevices, err := container.AllLinkLayerDevices()
 	if err != nil {
-		return err
+		return errors.Trace(err)
 	}
 
+	// We do not ask the provider to allocate addresses for manually provisioned
+	// machines as we do not expect such machines to be recognised (LP:1796106).
+	hostIsManual, err := host.IsManual()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	supportContainerAddresses := environs.SupportsContainerAddresses(callContext, env) && !hostIsManual
```

Comment:
shouldn't the hostIsManual check come before the environs.Supports check, so that we don't do the environs call if we know the bool?

supportsContainerAddresses := hostIsManual
if !hostIsManual {
  supportsContainerAddresses = environsSupports...
}

That said, 'supportContainerAddresses' no longer means what it says. It actually means:

askProviderForAddress := ...

I think switching the name will be clearer when debugging issues in the future.